src/txDigester/txDigests.ts

```diff
@@ -0,0 +1,76 @@
+import * as db from './sqlite3storage'
+import { digesterDatabase, extractValues } from './sqlite3storage'
+import { config } from '../Config'
+
+/**
+ * TransactionDigest is for storing transaction digests, which is the hash of prevHash
+ * and list of transactions in a timestamp range
+ */
+export interface TransactionDigest {
+  cycleStart: number
+  cycleEnd: number
+  txCount: number
+  hash: string
+}
+
+export async function insertTransactionDigest(txDigest: TransactionDigest): Promise<void> {
+  try {
+    const fields = Object.keys(txDigest).join(', ')
+    const placeholders = Object.keys(txDigest).fill('?').join(', ')
+    const values = extractValues(txDigest)
+    const sql = 'INSERT OR REPLACE INTO txDigests (' + fields + ') VALUES (' + placeholders + ')'
```

Comment:
same suggestion as above for using `ON CONFLICT` clause

Reply:
Done